packages/app-store/src/appStore.ts

```diff
@@ -60,6 +60,32 @@ export type CustomSetState = (
 
 export type SetterCreator<K extends keyof AppState> = (setState: CustomSetState) => AppState[K]
 
+/**
+ * Cleans up title field in save list items by extracting the filename without extension
+ * when the title is actually a full path
+ */
+const cleanupSaveListTitles = (saveList: SaveItem[]): SaveItem[] => {
+  let cleanedList = saveList.map((item) => {
+    // Check if the title looks like a path
+    if (item.title && (item.title.includes('/') || item.title.includes('\\'))) {
+      // Extract filename without extension from the path
+      const filename = item.title.split(/[\/\\]/).pop() || item.title
+      // Remove extension if present
+      const titleWithoutExt = filename.replace(/\.[^/.]+$/, '')
+      return { ...item, title: titleWithoutExt }
+    }
+    return item
+  })
```

Comment:
So I think you're trying to solve this problem in the wrong place. 

This cleanup shouldn't be necessary as we should always be extracting this filename whenever the user saves a file (see `packages/desktop/src/renderer/handlers/fileHandlers.ts` line 87). I don't have that problem on my mac, however clearly something is broken on Windows...

We're using `path.basename(response.savePath)` to grab that filename, and as an improvement we can even do `path.basename(response.savePath, '.json')` to remove the `.json` extension. This _should_ be OS specific, and work across both Windows and POSIX systems. Not exactly sure what's going on here??